### ISSUE TITLE
Dlt-daemon problem with embedded platform cross-compilation

### ISSUE THREAD

**Issue by Gonglja** [GENIVI/dlt-daemon#237]
Hello Developer,
I am a system integration engineer. When I use dlt-daemon, it can be compiled normally under the x86_64 platform and NXP imx8mm, but the compilation under NXP s32v234 and nvidia's pegasus reports errors (the latter three are all used under ubuntu1604 Cross-compilation toolchain compilation),

build step：
```
mkdir build && cd build 
cmake -DCMAKE_C_COMPILER=XXX/gcc -DCMAKE_CPP+COMPILER=XXX/g++ ..
make 
```
Error is as follows，
can you help me,thanks
```
/home/ubuntu/dlt-daemon/src/lib/dlt_user.c: In function ‘dlt_start_threads’:
/home/ubuntu/dlt-daemon/src/lib/dlt_user.c:4635:34: error: ‘DltUser {aka struct <anonymous>}’ has no member named ‘dlt_segmented_nwt_handle’; did you mean ‘dlt_log_handle’?
     if (pthread_create(&(dlt_user.dlt_segmented_nwt_handle), NULL,
                                  ^
/home/ubuntu/dlt-daemon/src/lib/dlt_user.c:4636:32: error: ‘dlt_user_trace_network_segmented_thread’ undeclared (first use in this function)
                        (void *)dlt_user_trace_network_segmented_thread, NULL)) {
```
```
ubuntu@ubuntu:build$ cmake -DCMAKE_C_COMPILER=/opt/gcc-linaro-6.3.1-2017.05-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-gcc -DCMAKE_CXX_COMPILER=/opt/gcc-linaro-6.3.1-2017.05-x86_64_aarch64-linux-gnu/b 
in/aarch64-linux-gnu-g++ ..
-- The C compiler identification is GNU 6.3.1
-- The CXX compiler identification is GNU 6.3.1
-- Check for working C compiler: /opt/gcc-linaro-6.3.1-2017.05-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-gcc
-- Check for working C compiler: /opt/gcc-linaro-6.3.1-2017.05-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/gcc-linaro-6.3.1-2017.05-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-g++
-- Check for working CXX compiler: /opt/gcc-linaro-6.3.1-2017.05-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build type not defined. Using default build type 'RelWithDebInfo'.
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Looking for include file arpa/inet.h
-- Looking for include file arpa/inet.h - found
-- Looking for include file fcntl.h
-- Looking for include file fcntl.h - found
-- Looking for include file float.h
-- Looking for include file float.h - found
-- Looking for include file limits.h
-- Looking for include file limits.h - found
-- Looking for include file netdb.h
-- Looking for include file netdb.h - found
-- Looking for include file netinet/in.h
-- Looking for include file netinet/in.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file sys/ioctl.h
-- Looking for include file sys/ioctl.h - found
-- Looking for include file sys/socket.h
-- Looking for include file sys/socket.h - found
-- Looking for include file sys/time.h
-- Looking for include file sys/time.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file sys/ipc.h
-- Looking for include file sys/ipc.h - found
-- Looking for include file ctype.h
-- Looking for include file ctype.h - found
-- Looking for include file signal.h
-- Looking for include file signal.h - found
-- Looking for include file syslog.h
-- Looking for include file syslog.h - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file linux/stat.h
-- Looking for include file linux/stat.h - found
-- Looking for include file sys/uio.h
-- Looking for include file sys/uio.h - found
-- Looking for include file termios.h
-- Looking for include file termios.h - found
-- Looking for include file mqueue.h
-- Looking for include file mqueue.h - found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for floor
-- Looking for floor - not found
-- Looking for fork
-- Looking for fork - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Looking for inet_ntoa
-- Looking for inet_ntoa - found
-- Looking for malloc
-- Looking for malloc - found
-- Looking for memmove
-- Looking for memmove - found
-- Looking for memset
-- Looking for memset - found
-- Looking for mkfifo
-- Looking for mkfifo - found
-- Looking for select
-- Looking for select - found
-- Looking for socket
-- Looking for socket - found
-- Looking for strchr
-- Looking for strchr - found
-- Looking for strerror
-- Looking for strerror - found
-- Looking for strstr
-- Looking for strstr - found
-- Looking for strtol
-- Looking for strtol - found
-- Looking for mq_open
-- Looking for mq_open - not found
-- Looking for mq_close
-- Looking for mq_close - not found
-- Looking for mq_unlink
-- Looking for mq_unlink - not found
-- Looking for mq_send
-- Looking for mq_send - not found
-- Looking for mq_receive
-- Looking for mq_receive - not found
-- Disable network trace interface since message queue is not supported
-- pthread_setname_np API not available on this platform
-- 
-- -------------------------------------------------------------------------------
-- Build for Version 2.18.5 build v2.18.5 version state STABLE
-- WITH_SYSTEMD = OFF
-- WITH_SYSTEMD_WATCHDOG = OFF
-- WITH_SYSTEMD_JOURNAL = OFF
-- WITH_DOC = OFF
-- WITH_MAN = OFF
-- WITH_DLT_ADAPTOR = OFF
-- WITH_DLT_CONSOLE = ON
-- WITH_DLT_EXAMPLES = ON
-- WITH_DLT_SYSTEM = OFF
-- WITH_DLT_FILETRANSFER = OFF
-- WITH_DLT_DBUS = OFF
-- WITH_DLT_TESTS = ON
-- WITH_DLT_UNIT_TESTS = OFF
-- WITH_DLT_SHM_ENABLE = OFF
-- WITH_DLTTEST = OFF
-- WITH_DLT_PKGCONFIG = ON
-- WITH_DLT_CXX11_EXT = OFF
-- WITH_DLT_COREDUMPHANDLER = OFF
-- WITH_DLT_KPI = OFF
-- WITH_DLT_FATAL_LOG_TRAP = OFF
-- WITH_CHECK_CONFIG_FILE = OFF
-- WITH_TESTSCRIPTS = OFF
-- WITH_GPROF = OFF
-- WITH_DLT_USE_IPv6 = ON
-- DLT_USER = genivi
-- BUILD_SHARED_LIBS = ON
-- TARGET_CPU_NAME = 
-- CMAKE_INSTALL_PREFIX = /usr/local
-- CMAKE_BUILD_TYPE = RelWithDebInfo
-- CMAKE_HOST_SYSTEM_PROCESSOR = x86_64
-- CMAKE_SYSTEM_PROCESSOR = x86_64
-- WITH_DLT_LOGSTORAGE_CTRL_UDEV = OFF
-- DLT_IPC = FIFO(Path: /tmp)
-- WITH_UDP_CONNECTION = OFF
-- WITH_DLT_QNX_SYSTEM = OFF)
-- Change a value with: cmake -D<Variable>=<Value>
-- -------------------------------------------------------------------------------
-- 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ubuntu/dlt-daemon/build
ubuntu@ubuntu:build$ make 
[  1%] Building C object src/lib/CMakeFiles/dlt.dir/dlt_user.c.o
/home/ubuntu/dlt-daemon/src/lib/dlt_user.c: In function ‘dlt_start_threads’:
/home/ubuntu/dlt-daemon/src/lib/dlt_user.c:4635:34: error: ‘DltUser {aka struct <anonymous>}’ has no member named ‘dlt_segmented_nwt_handle’; did you mean ‘dlt_log_handle’?
     if (pthread_create(&(dlt_user.dlt_segmented_nwt_handle), NULL,
                                  ^
/home/ubuntu/dlt-daemon/src/lib/dlt_user.c:4636:32: error: ‘dlt_user_trace_network_segmented_thread’ undeclared (first use in this function)
                        (void *)dlt_user_trace_network_segmented_thread, NULL)) {
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubuntu/dlt-daemon/src/lib/dlt_user.c:4636:32: note: each undeclared identifier is reported only once for each function it appears in
src/lib/CMakeFiles/dlt.dir/build.make:62: recipe for target 'src/lib/CMakeFiles/dlt.dir/dlt_user.c.o' failed
make[2]: *** [src/lib/CMakeFiles/dlt.dir/dlt_user.c.o] Error 1
CMakeFiles/Makefile2:174: recipe for target 'src/lib/CMakeFiles/dlt.dir/all' failed
make[1]: *** [src/lib/CMakeFiles/dlt.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by Gonglja** [comment]
DLT 2.18.4 can be programmed on both S32V and Pegasus platforms

**Comment by ssugiura** [comment]
@Gonglja This issue comes from the commit https://github.com/GENIVI/dlt-daemon/commit/404bc3934bcb0bbb5671a84ecb45c95069bc1ab3. We will take a look into it and let you know the root cause and how you can fix it.

**Comment by Kevin-Luong** [comment]
Hello @Gonglja 

I can reproduce your issue and then I have tried the following command and get the build success result:

lko1hc@ :~/workspace/dlt/dlt-daemon/build$ cmake -DCMAKE_C_COMPILER=/opt/gcc-linaro-7.5.0-2019.12-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-gcc -DCMAKE_CXX_COMPILER=/opt/gcc-linaro-7.5.0-2019.12-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-g++ -DDLT_NETWORK_TRACE_ENABLE=ON .. ; make

Hope this temporary solution can solve your issue.

**Comment by Gonglja** [comment]
> 你好@Gonglja
> 
> 我可以重现您的问题，然后我已尝试以下命令，并获得生成成功结果：
> 
> lko1hc@：\/工作区/dlt/dlt-daemon/build$ cmake -DCMAKE_C_COMPILER\/opt/gcc-linaro-7.5.0-2019.12-x86_64_aarch64-linux-gnu/bin/aarch64-linux - gnu - gcc - DCMAKE_CXX_COMPILER=/opt/gcc-linaro-7.5.0-2019.12-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-g= -DDLT_NETWORK_TRACE_ENABLE=ON.;让
> 
> 希望这个临时解决方案能解决您的问题。

Thank you very much, the temporary solution (-DDLT_NETWORK_TRACE_ENABLE=ON) has been compiled successfully.

**Comment by ssugiura** [comment]
@Gonglja We've solved the issue with https://github.com/GENIVI/dlt-daemon/pull/239.
The issue was that the compiler couldn't find message queue interfaces, and it will disable network trace feature.
However some code was left over.
Hope this helps to proceed. Sorry for inconvenience caused and thank you!

**Comment by Gonglja** [comment]
> @Gonglja We've solved the issue with #239.
> The issue was that the compiler couldn't find message queue interfaces, and it will disable network trace feature.
> However some code was left over.
> Hope this helps to proceed. Sorry for inconvenience caused and thank you!

Yes, I have received it, I tried to compile the new version and it has passed, thank you very much
